src/core_codemods/remove_unused_imports.py

```diff
@@ -111,25 +119,42 @@ def _is_disabled_by_linter(self, node: cst.CSTNode) -> bool:
         return False
 
 
+class GatherCommentNodes(CSTVisitor):
+    def __init__(self) -> None:
+        self.comments: list[cst.Comment] = []
+        super().__init__()
+
+    def leave_Comment(self, original_node: cst.Comment) -> None:
+        self.comments.append(original_node)
+
+
 def match_line(pos, line):
     return pos.start.line == line and pos.end.line == line
 
 
 def _is_pylint_disable_unused_imports(comment: str) -> bool:
-    parsed = parse_pragma(comment)
-    for p in parsed:
-        if p.action == "disable" and (
-            "unused-import" in p.messages or "W0611" in p.messages
-        ):
-            return True
+    # If pragma parse fails, ignore
+    try:
+        parsed = parse_pragma(comment)
+        for p in parsed:
+            if p.action == "disable" and (
+                "unused-import" in p.messages or "W0611" in p.messages
+            ):
+                return True
+    except Exception:
+        pass
     return False
 
 
 def _is_pylint_disable_next_unused_imports(comment: str) -> bool:
-    parsed = parse_pragma(comment)
-    for p in parsed:
-        if p.action == "disable-next" and (
-            "unused-import" in p.messages or "W0611" in p.messages
-        ):
-            return True
+    # If pragma parse fails, ignore
```

Comment:
Do we have a test for this case? It would be useful to add the case that caused the error in the first place.

Reply:
It was failing because of a "# NOQA" comment in one of the files in `pytest`. The thing is, any pragma that is not recognized by `pylint` will raise an exception (note that we use a function from `pylint` directly). We have no control of those, hence the catch-all `try`-`except`.

Reply:
We should probably honor `# noqa` as well for this codemod but that doesn't have to happen in this PR.

Reply:
We do (and have tests for it). I've also fixed an issue in this PR where the regular expression that detects the `# noqa` pragma was case sensitive.